docs/docs/guides/evaluation/imperative_evaluations.md

```diff
@@ -0,0 +1,83 @@
+In contrast to the standard `Evaluation` object, which requires a predefined `Dataset` and list of `Scorer` objects, the imperative logger allows you to log individual predictions and their associated scores incrementally as they become available.
+
+:::info Looking for a more opinionated approach?
+
+If you prefer a more structured evaluation framework with predefined datasets and scorers, check out Weave's standard [Evaluation framework](../core-types/evaluations.md). The standard approach provides a more declarative way to define and run evaluations, with built-in support for datasets, scorers, and comprehensive reporting.
+
+The imperative approach described on this page offers more flexibility for complex workflows, while the standard evaluation framework provides more structure and guidance.
+
+:::
+
+## Basic Usage
+
+1.  **Initialize the logger:** Create an instance of `ImperativeEvaluationLogger`. You can optionally provide strings or dictionaries as metadata for the `Model` and `Dataset` being evaluated. If omitted, default placeholders are used.
+2.  **Log Predictions:** For each input/output pair from your model or system, call `log_prediction`. This method returns an `ImperativeScoreLogger` object tied to that specific prediction event.
+3.  **Log Scores:** Use the `ImperativeScoreLogger` object obtained in the previous step to log scores via the `log_score` method. You can log multiple scores from different conceptual scorers (identified by string names or `Scorer` objects) for the same prediction. Call `finish()` on the score logger when you're done logging scores for that prediction to finalize it. _Note: After calling `finish()`, the `ImperativeScoreLogger` instance cannot be used to log additional scores._
+4.  **Log Summary:** After processing all your examples and logging their predictions and scores, call `log_summary` on the main `ImperativeEvaluationLogger` instance. This action finalizes the overall evaluation. Weave automatically calculates summaries for common score types (like counts and fractions for boolean scores) and merges these with any custom summary dictionary you provide. You can include metrics not logged as row-level scores, such as total elapsed time or other aggregate measures, in this summary dictionary.
+
+## Example
+
+```python
+import weave
+from openai import OpenAI
+from weave.flow.eval_imperative import ImperativeEvaluationLogger
+
+# Initialize the logger (model/dataset names are optional metadata)
+eval_logger = ImperativeEvaluationLogger(
+    model="my-model",
+    dataset="my-dataset"
+)
+
+# Example input data (this can be any data structure you want)
+eval_samples = [
+    {'inputs': {'a': 1, 'b': 2}, 'expected': 3},
+    {'inputs': {'a': 2, 'b': 3}, 'expected': 5},
+    {'inputs': {'a': 3, 'b': 4}, 'expected': 7},
+]
+
+# Example model logic.  This does not have to be decorated with @weave.op,
+# but if you do, it will be traced and logged.
+@weave.op
+def user_model(a: int, b: int) -> int:
```

Comment:
Is this supposed to be `use_model`?

Reply:
no, it's the user's model.  Maybe I can rename to `user_defined_model`